tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -66,41 +66,49 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
-        EXPECT_NOT_NULL(verifying_conn = s2n_connection_new(verifier_mode));
-        verifying_conn->actual_protocol_version = S2N_TLS13;
-        EXPECT_NOT_NULL(sending_conn = s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT));
-
-        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_in, S2N_MAX_TEST_PEM_SIZE));
-        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_out, S2N_MAX_TEST_PEM_SIZE));
-
-        EXPECT_SUCCESS(s2n_connection_set_config(sending_conn, config));
+        DEFER_CLEANUP(struct s2n_connection *sending_conn =
+                              s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(sending_conn);
         sending_conn->handshake_params.our_chain_and_key = cert_chain;
         sending_conn->handshake_params.server_cert_sig_scheme = &sig_scheme;
         sending_conn->handshake_params.client_cert_sig_scheme = &sig_scheme;
         sending_conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
         sending_conn->actual_protocol_version = S2N_TLS13;
+        EXPECT_SUCCESS(s2n_connection_set_config(sending_conn, config));
 
-        EXPECT_SUCCESS(s2n_connection_set_config(verifying_conn, config));
+        DEFER_CLEANUP(struct s2n_connection *verifying_conn = s2n_connection_new(verifier_mode),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(verifying_conn);
         verifying_conn->secure->cipher_suite = &s2n_tls13_aes_128_gcm_sha256;
+        verifying_conn->actual_protocol_version = S2N_TLS13;
+        EXPECT_SUCCESS(s2n_connection_set_config(verifying_conn, config));
 
         EXPECT_SUCCESS(s2n_blob_init(&b, (uint8_t *) cert_chain_pem, strlen(cert_chain_pem) + 1));
+        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_in, S2N_MAX_TEST_PEM_SIZE));
+        EXPECT_SUCCESS(s2n_stuffer_alloc(&certificate_out, S2N_MAX_TEST_PEM_SIZE));
         EXPECT_SUCCESS(s2n_stuffer_write(&certificate_in, &b));
         EXPECT_SUCCESS(s2n_stuffer_certificate_from_pem(&certificate_in, &certificate_out));
 
         /* Extract public key from certificate and set it for verifying connection */
         uint32_t available_size = s2n_stuffer_data_available(&certificate_out);
-        EXPECT_SUCCESS(s2n_blob_init(&b, s2n_stuffer_raw_read(&certificate_out, available_size), available_size));
+        EXPECT_SUCCESS(s2n_blob_init(&b, s2n_stuffer_raw_read(&certificate_out, available_size),
+                available_size));
         if (verifying_conn->mode == S2N_CLIENT) {
-            EXPECT_OK(s2n_asn1der_to_public_key_and_type(&verifying_conn->handshake_params.server_public_key, &pkey_type, &b));
-            EXPECT_SUCCESS(s2n_pkey_match(&verifying_conn->handshake_params.server_public_key, sending_conn->handshake_params.our_chain_and_key->private_key));
+            EXPECT_OK(s2n_asn1der_to_public_key_and_type(&verifying_conn->handshake_params.server_public_key,
```

Comment:
We have to instantiate server/client_public_key per test case with the`b` variable. However, all the code that we're using to instantiate `b` is a lot of repetitive code that actually only needs to be done once, in the beginning of the test file, essentially because the certificate we're using doesn't change per test case. 

Can you move that code up to the beginning of run_tests()?